bundles/AdminBundle/Controller/AdminController.php

```diff
@@ -1,337 +1,198 @@
-    {
-        /** @var SerializerInterface|DecoderInterface $serializer */
-        $serializer = null;
-
-        if ($useAdminSerializer) {
-            $serializer = $this->container->get('pimcore_admin.serializer');
-        } else {
-            $serializer = $this->container->get('serializer');
-        }
-
-        if ($associative) {
-            $context['json_decode_associative'] = true;
-        }
-
-        return $serializer->decode($json, 'json', $context);
-    }
-
-    /**
-     * Returns a JsonResponse that uses the admin serializer
-     *
-     * @param mixed $data    The response data
-     * @param int $status    The status code to use for the Response
-     * @param array $headers Array of extra headers to add
-     * @param array $context Context to pass to serializer when using serializer component
-     * @param bool $useAdminSerializer
-     *
-     * @return JsonResponse
-     */
-    protected function adminJson($data, $status = 200, $headers = [], $context = [], bool $useAdminSerializer = true)
-    {
-        $json = $this->encodeJson($data, $context, JsonResponse::DEFAULT_ENCODING_OPTIONS, $useAdminSerializer);
-
-        return new JsonResponse($json, $status, $headers, true);
-    }
-
-    /**
-     * Translates the given message.
-     *
-     * @param string $id The message id (may also be an object that can be cast to string)
-     * @param array $parameters An array of parameters for the message
-     * @param string|null $domain The domain for the message or null to use the default
-     * @param string|null $locale The locale or null to use the default
-     *
-     * @return string The translated string
-     *
-     * @throws InvalidArgumentException If the locale contains invalid characters
-     */
-    public function trans($id, array $parameters = [], $domain = 'admin', $locale = null)
-    {
-        return $this->translator->trans($id, $parameters, $domain, $locale);
-    }
-}
+<?php
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\AdminBundle\Controller;
+
+use Pimcore\Bundle\AdminBundle\HttpFoundation\JsonResponse;
+use Pimcore\Bundle\CoreBundle\Controller\UserAwareController;
+use Pimcore\Extension\Bundle\PimcoreBundleManager;
+use Pimcore\Security\User\TokenStorageUserResolver;
+use Symfony\Component\Serializer\Encoder\DecoderInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * @deprecated and will be removed in Pimcore 11. Use Pimcore\Bundle\CoreBundle\Controller\UserAwareController instead.

```

Comment:
Should we also change internal classes already not to extend from deprecated class?

Reply:
👍 yep

Reply:
but then admin serializer will not be used anymore for the internal calls. right? this could be a breaking change then?

Reply:
maybe we could just introduce a new abstract admin controller(internal) which still uses admin serializer and then we switch to pimcore serializer on Pimcore 11. wdyt?